### ISSUE TITLE
Assets are packed inside exe if PublishSingleFile is used (different from .NET 8.0 Console app)

### ISSUE THREAD

**Issue by Prochy** [microsoft/WindowsAppSDK#4242]
### Describe the bug

Not 100% sure if it's intended behavior or if it's a bug, but anyway it behaves different from .NET8.0 Console application if I use PublishSingleFile=true.

If I use PublishSingleFile=true in Console App in NET 8.0, next to the single file, there files that have attributes Content with Copy always to output folder. 
If I use the same publish command with WinUI Net 8.0, next to the single file, there are no files. The files are then in temp directory where the app is unpacked.

This causes me troubles as e.g. if I have this command to construct Configuration root, it fails that it cannot find the file.
```
var conf = new ConfigurationBuilder()
    .SetBasePath(workDir)
    .AddJsonFile("appsettings.json")
    .Build();
```

### Steps to reproduce the bug

Create WinUI3 project and add there a asset file properties = "Content":"Copy always to output directory"
Build project with 
`dotnet publish -p:PublishSingleFile=true --output publish -p:RuntimeIdentifier=win-x64 -p:Platform=x64 -p:WindowsAppSDKSelfContained=true`

### Expected behavior

The file should be placed next to the single file.

### NuGet package version

Windows App SDK 1.5.0: 1.5.240227000

### Packaging type

Unpackaged

### Windows version

Windows 10 version 21H2 (19044, November 2021 Update)

### IDE

Visual Studio 2022

### Additional context

As I said not sure if it is intended or not but in my opinion I would guess the consistency should be kept across different types of .net applications. If it is intended, you can reject it.
I know I can fix it for exampe by explicitly setting path to the temp directory by AppDomain.CurrentDomain.BaseDirectory but I expect the file to be located inside working directory which should be the folder with singlefile exe and not the temp directory.

**Edit note**
I think also for some changing values in the appsettings.json file (enabling debug log or other changes), it's better when those files are next to the single file. For now (until I get info if it's on purpose or not), I solved this problem by extra post publish step if someone is intrested:

```
	<Target Name="CopyContentFilesAfterPublish" AfterTargets="Publish">
		<Exec Command="move &quot;$(TargetDir)\Assets&quot; &quot;$(PublishDir)&quot;" />
	</Target>
```

**Comment by DarranRowe** [comment]
To note something important, it is always problematic to rely on the current working directory in your application. To give a little example of this, consider the following code.

```
#include <filesystem>
#include <iostream>

int wmain()
{
	auto cwd = std::filesystem::current_path();
	std::wcout << cwd << L"\n";

	return 0;
}
```

This obtains the current working directory and then prints it out to the console.

![Screenshot 2024-03-02 171224](https://github.com/microsoft/WindowsAppSDK/assets/52577874/ac3a2330-8e2e-43d8-a4e0-455fc5605bb5)

This is the output from running the application. Do you notice the interesting thing in the output? Yes, it prints the same directory as the current directory in the console window.
Additionally, consider the shortcut properties window.

![Screenshot 2024-03-02 171459](https://github.com/microsoft/WindowsAppSDK/assets/52577874/dba6c060-fcd8-457f-a87c-6ae297dd8249)

The "Start in" value corresponds to the initial working directory for the application.
Finally, consider the [start](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/start) command. This has the /D option which allows the user to directly set the working directory of the application.
In other words, the user can directly affect the working directory of the application which means that it is an unreliable path.

**Comment by ghost1372** [comment]
use 
`<ExcludeFromSingleFile>true</ExcludeFromSingleFile>
`

example:

```
<ItemGroup>
  <Content Update="Plugin.dll">
    <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
    <ExcludeFromSingleFile>true</ExcludeFromSingleFile>
  </Content>
</ItemGroup>
```

**Comment by Scottj1s** [comment]
@Prochy Yes, this behavior difference is intentional.  Windows App SDK's PublishSingleFile support relies on SxS DLL redirection, which in turn requires all content to be extracted.  This is done by automatically setting IncludeAllContentForSelfExtract=true, which you can see in a binlog.  If you set that explicitly in your console app, you'll see the same behavior.  The workaround is the fine-grained ExcludeFromSingleFile, mentioned by @ghost1372.
